docs/plotting.md

```diff
@@ -0,0 +1,97 @@
+# Recommended interface and behavior of a `plot` method
```

Comment:
You should create a `docs/dev` directory and put this inside of it, since this documentation is intended for developers and not users. We can also move `development.md` into it (but be careful to update any references in sphinx configuration).